### ISSUE TITLE
Don't NoWarn CS1591 in ProjectDefaults.props

### ISSUE THREAD

**Issue by bricelam** [dotnet/arcade#2594]
CS1591 (Missing XML comment for publicly visible type or member) is disabled in ProjectDefaults.props

https://github.com/dotnet/arcade/blob/851e36df83d3361e4bd8a70a2a8a89f762469f9a/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectDefaults.props#L90

Most repos *should* have XML comments on publicly visible types and members, so suppressing this in ProjectDefaults.props seems wrong.

Also, it would be a lot easier for repos to suppress this warning than it is to re-enable it.

Our current workaroud:

``` xml
<ItemGroup>
  <_NoWarns Include="$(NoWarn)" Exclude="1591;CS1591" />
</ItemGroup>
<PropertyGroup>
  <NoWarn>@(_NoWarns)</NoWarn>
</PropertyGroup>
```

cc @ajcvickers

**Comment by tmat** [comment]
I disagree. This warning is gonna produce a lot of noise in most repos and shouldn't have been implemented as a compiler warning in the first place, IMO. The problem is that the compiler doesn't have enough context to determine that a public type/member is a shipping public API that should be documented. There are many cases (e.g. test assemblies, test libraries, etc.) when doc commends do not add any value, quite contrary they are just clutter (e.g. like comment "Gets a foo" on method `GetFoo` that's there just to shut the warning up).

That said, enforcing doc comments on shipping public APIs would be useful. We do have a [public API analyzer](https://github.com/dotnet/roslyn-analyzers/tree/master/src/PublicApiAnalyzers) and it would be useful to implement this functionality in that analyzer, which has the full context.

**Comment by markwilkie** [comment]
Thoughts @bricelam ?

**Comment by bricelam** [comment]
We've worked around it in our repo. I just filed this issue since it felt like a bad default (i.e. a pit of failure).

We have a team policy of documenting all public APIs and this (along with a handful of [StyleCop rules](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/DocumentationRules.md)) help enforce that. We had a few undocumented APIs slip through before we noticed this was being disabled by Arcade.
